network/loadbalancer.go

```diff
@@ -0,0 +1,18 @@
+package network
+
+import (
+	gerr "github.com/gatewayd-io/gatewayd/errors"
+)
+
+type LoadBalancerStrategy interface {
+	GetNextProxy() IProxy
```

Comment:
Nit: `Get` is not needed.

```suggestion
	NextProxy() IProxy
```